### PR TITLE
Ensure exit code 130 is returned from SIGINT

### PR DESCRIPTION
**Pull request by b-deam**
With this commit we ensure that rally will return
exit code 130 when its execution is interrupted via CTRL + C / SIGINT.

Closes https://github.com/elastic/rally/issues/1222

### CODE REVIEW

File: esrally/racecontrol.py [hunk 1]
```diff
@@ -256,6 +256,7 @@ def race(cfg, sources=False, distribution=False, external=False, docker=False):
         # notify the coordinator so it can properly handle this state. Do it blocking so we don't have a race between this message
         # and the actor exit request.
         actor_system.ask(benchmark_actor, actor.BenchmarkCancelled())
+        raise exceptions.UserInterrupted("User has cancelled the benchmark (detected by race control).") from None
```
Comment (danielmitterdorfer) [inline esrally/racecontrol.py:259]: There is one more place in the `racecontrol` module where we handle `KeyboardInterrupt`; we also need to raise `exceptions.UserInterrupted` there.

**Review by danielmitterdorfer — COMMENTED** [review]
Thanks for your PR! I left a couple of suggestions. 

It's only semi-related to your PR but during testing I found a case where we swallow SIGINT, namely in `esrally.net.has_internet_connection`. We wrap probing in a `try ... except BaseException` (which includes `KeyboardInterrupt`) and mistakenly print "No Internet connection detected ..." and continue with the benchmark.

If we handle this specifically and re-raise the exception, Rally aborts:

```
    except KeyboardInterrupt:
        raise
```

Even with this we don't get exit code 130 but IMHO that's fine in the initialization phase. If we hit Ctrl+C even earlier then we get e.g. stack traces from Python's module initialization code and it's unlikely - and also ok - that we're able to consistently handle errors that early during application initialization.

**Comment by b-deam** [comment]
> Thanks for your PR! I left a couple of suggestions.
> 
> It's only semi-related to your PR but during testing I found a case where we swallow SIGINT, namely in `esrally.net.has_internet_connection`. We wrap probing in a `try ... except BaseException` (which includes `KeyboardInterrupt`) and mistakenly print "No Internet connection detected ..." and continue with the benchmark.
> 
> If we handle this specifically and re-raise the exception, Rally aborts:
> 
> ```
>     except KeyboardInterrupt:
>         raise
> ```
> 
> Even with this we don't get exit code 130 but IMHO that's fine in the initialization phase. If we hit Ctrl+C even earlier then we get e.g. stack traces from Python's module initialization code and it's unlikely - and also ok - that we're able to consistently handle errors that early during application initialization.

Interestingly I _do_ get 130 as an exit code after implementing the `KeyboardInterrupt` exception handler:
```
$ esrally race --distribution-version=6.5.3 --track=geonames --kill-running-processes

    ____        ____
   / __ \____ _/ / /_  __
  / /_/ / __ `/ / / / / /
 / _, _/ /_/ / / / /_/ /
/_/ |_|\__,_/_/_/\__, /
                /____/

^CTraceback (most recent call last):
  File "/Users/bradleydeam/perf/github.com/b-deam/rally/.venv/bin/esrally", line 33, in <module>
    sys.exit(load_entry_point('esrally', 'console_scripts', 'esrally')())
  File "/Users/bradleydeam/perf/github.com/b-deam/rally/esrally/rally.py", line 946, in main
    if not net.has_internet_connection(probing_url):
  File "/Users/bradleydeam/perf/github.com/b-deam/rally/esrally/utils/net.py", line 252, in has_internet_connection
    response = __http().request("GET", probing_url, timeout=2.0)
  File "/Users/bradleydeam/perf/github.com/b-deam/rally/.venv/lib/python3.8/site-packages/urllib3/request.py", line 75, in request
    return self.request_encode_url(
  File "/Users/bradleydeam/perf/github.com/b-deam/rally/.venv/lib/python3.8/site-packages/urllib3/request.py", line 97, in request_encode_url
    return self.urlopen(method, url, **extra_kw)
  File "/Users/bradleydeam/perf/github.com/b-deam/rally/.venv/lib/python3.8/site-packages/urllib3/poolmanager.py", line 336, in urlopen
    response = conn.urlopen(method, u.request_uri, **kw)
  File "/Users/bradleydeam/perf/github.com/b-deam/rally/.venv/lib/python3.8/site-packages/urllib3/connectionpool.py", line 690, in urlopen
    response = self.ResponseCls.from_httplib(
  File "/Users/bradleydeam/perf/github.com/b-deam/rally/.venv/lib/python3.8/site-packages/urllib3/response.py", line 600, in from_httplib
    resp = ResponseCls(
  File "/Users/bradleydeam/perf/github.com/b-deam/rally/.venv/lib/python3.8/site-packages/urllib3/response.py", line 258, in __init__
    self._body = self.read(decode_content=decode_content)
  File "/Users/bradleydeam/perf/github.com/b-deam/rally/.venv/lib/python3.8/site-packages/urllib3/response.py", line 514, in read
    data = self._fp.read() if not fp_closed else b""
  File "/Users/bradleydeam/.pyenv/versions/3.8.8/lib/python3.8/http/client.py", line 465, in read
    return self._readall_chunked()
  File "/Users/bradleydeam/.pyenv/versions/3.8.8/lib/python3.8/http/client.py", line 575, in _readall_chunked
    value.append(self._safe_read(chunk_left))
  File "/Users/bradleydeam/.pyenv/versions/3.8.8/lib/python3.8/http/client.py", line 612, in _safe_read
    data = self.fp.read(amt)
  File "/Users/bradleydeam/.pyenv/versions/3.8.8/lib/python3.8/socket.py", line 669, in readinto
    return self._sock.recv_into(b)
  File "/Users/bradleydeam/.pyenv/versions/3.8.8/lib/python3.8/ssl.py", line 1241, in recv_into
    return self.read(nbytes, buffer)
  File "/Users/bradleydeam/.pyenv/versions/3.8.8/lib/python3.8/ssl.py", line 1099, in read
    return self._sslobj.read(len, buffer)
KeyboardInterrupt

(.venv) bradleydeam in ~/perf/github.com/b-deam/rally on branch exit-code-update $ echo $?
130
```

But agreed nonetheless with your previous comments.


I've made the requested changes, can you take a look?

**Comment by danielmitterdorfer** [comment]
The test failure appears to be unrelated and transient.

@elasticmachine test this please

**Review by danielmitterdorfer — APPROVED** [review]
Thanks for iterating on this; it works fine now. LGTM

**Pull request merged**